src/Type/Php/CtypeDigitFunctionTypeSpecifyingExtension.php

```diff
@@ -54,7 +56,18 @@ public function specifyTypes(FunctionReflection $functionReflection, FuncCall $n
 			]);
 		}
 
-		return $this->typeSpecifier->create($node->getArgs()[0]->value, TypeCombinator::union(...$types), $context, false, $scope);
+		$unionType = TypeCombinator::union(...$types);
+		$specifiedTypes = $this->typeSpecifier->create($exprArg, $unionType, $context, false, $scope);
+
+		if ($context->true() && $exprArg instanceof Cast) {
```

Comment:
my initial attempt was to do this directly within the `TypeSpecifier` ... but I then got the feeling that the type narrowing of a cast-expression is not a concept which can be applied to all specifying extensions.

Reply:
Yeah, well, `is_string((string) $int)` is always true, but it doesn't make `$int` a string :D